manifests/base/repo-server/kustomization.yaml

```diff
@@ -0,0 +1,7 @@
+commonLabels:
+  app.kubernetes.io/component: repo-server
+  app.kubernetes.io/name: argo-cd
```

Comment:
I actually dislike the behavior which kustomize's has regarding commonLabels, which injects into the `spec.template.metadata.labels` since I believe `spec.selector.matchLabels` should be carefully decided and controlled, and are often times intentionally different than the top level `metadata.labels`. For example, it is forcing us to use the component name as the selector, when we currently use the deployment name as the selector.

It results in a manifest like:
```
metadata:
  labels:
    app.kubernetes.io/component: application-controller
    app.kubernetes.io/name: argo-cd
  name: argocd-application-controller
spec:
  selector:
    matchLabels:
      app.kubernetes.io/component: application-controller
      app.kubernetes.io/name: argo-cd
  template:
    metadata:
      labels:
        app.kubernetes.io/component: application-controller
        app.kubernetes.io/name: argo-cd
```

I think we should only add the labels to the top level resources, which would result on a manifest like:

```
metadata:
  labels:
    app.kubernetes.io/component: application-controller
    app.kubernetes.io/name: argo-cd
  name: argocd-application-controller
spec:
  selector:
    matchLabels:
      argocd-application-controller
  template:
    metadata:
      labels:
        argocd-application-controller
```

This probably means we cannot use Kustomize's commonLabels feature.

Reply:
I *think* I'm ok with not using commonLabels and hard-coding the `app.kubernetes.io` labels as you describe. My only reservation is that if those labels are meant to be used for dashboards, not adding `app.kubernetes.io/name` as a Pod label means a dashboard couldn't show all Pods belonging to argo-cd.

Totally hypothetical at this time, but worth (re-)considering?

Reply:
True, but since kustomize is only additive, it's trivial to produce the manifests you desire with another kustomization, but non-trivial to remove the lablels with kustomize.

I rather wait to see how dashboards make use of these labels to go further. Dashboards will have the same problem that Argo CD used to have about pod association for objects *other* than Deployments, StatefulSets, DaemonSets, because kustomize will have no clue how to achieve the same label injection result with something like a Knative app. As we discovered, the only reliable way to know object association is via ownership references.